src/test/rustdoc/const-generics/const-impl.rs

```diff
@@ -33,7 +33,7 @@ impl<T> VSet<T, { Order::Unsorted }> {
 
 pub struct Escape<const S: &'static str>;
 
-// @has foo/struct.Escape.html '//div[@id="impl"]/h3[@class="code-header in-band"]' 'impl Escape<r#"<script>alert("Escape");</script>"#>'
+// @has foo/struct.Escape.html '//div[@id="impl-Escape%3Cr%23%22%3Cscript%3Ealert%28%22Escape%22%29%253B%3C%2Fscript%3E%22%23%3E"]/h3[@class="code-header in-band"]' 'impl Escape<r#"<script>alert("Escape");</script>"#>'
 impl Escape<r#"<script>alert("Escape");</script>"#> {
```

Comment:
These are copied from the resulting HTML files and then URL encoded... still not passing...

Reply:
Hmm, what do you mean "and then URL encoded"? I think just copying them straight from the HTML should work.

Reply:
I thought special characters need to be "percent encoded" here. Copying this string from the URL doesn't work.

Reply:
https://github.com/rust-lang/rust/pull/92745/commits/515eba72468aa30b27eb40bd17f5b550753d9650

Reply:
This diff gets the test to pass for me.

```diff
diff --git a/src/test/rustdoc/const-generics/const-impl.rs b/src/test/rustdoc/const-generics/const-impl.rs
index bd961fa6434..69a7cee71f3 100644
--- a/src/test/rustdoc/const-generics/const-impl.rs
+++ b/src/test/rustdoc/const-generics/const-impl.rs
@@ -1,7 +1,5 @@
 #![allow(incomplete_features)]
-
 #![feature(adt_const_params)]
-
 #![crate_name = "foo"]
 
 #[derive(PartialEq, Eq)]
@@ -17,14 +15,14 @@ pub struct VSet<T, const ORDER: Order> {
     inner: Vec<T>,
 }
 
-// @has foo/struct.VSet.html '//div[@id="impl-VSet<T%2C%20{%20Order%3A%3ASorted%20}>"]/h3[@class="code-header in-band"]' 'impl<T> VSet<T, { Order::Sorted }>'
+// @has foo/struct.VSet.html '//div[@id="impl-VSet%3CT%2C%20{%20Order%3A%3AUnsorted%20}%3E"]/h3[@class="code-header in-band"]' 'impl<T> VSet<T, { Order::Unsorted }>'
 impl<T> VSet<T, { Order::Sorted }> {
     pub fn new() -> Self {
         Self { inner: Vec::new() }
     }
 }
 
-// @has foo/struct.VSet.html '//div[@id="impl-VSet<T%2C%20{%20Order%3A%3AUnsorted%20}>"]/h3[@class="code-header in-band"]' 'impl<T> VSet<T, { Order::Unsorted }>'
+// @has foo/struct.VSet.html '//div[@id="impl-VSet%3CT%2C%20{%20Order%3A%3AUnsorted%20}%3E"]/h3[@class="code-header in-band"]' 'impl<T> VSet<T, { Order::Unsorted }>'
 impl<T> VSet<T, { Order::Unsorted }> {
     pub fn new() -> Self {
         Self { inner: Vec::new() }
@@ -33,7 +31,7 @@ pub fn new() -> Self {
 
 pub struct Escape<const S: &'static str>;
 
-// @has foo/struct.Escape.html '//div[@id="impl-Escape<r#"<script>alert("Escape")%3B</script>"#>"]/h3[@class="code-header in-band"]' 'impl Escape<r#"<script>alert("Escape");</script>"#>'
+// has foo/struct.Escape.html '//div[@id="impl-Escape<r#"<script>alert("Escape")%3B</script>"#>"]/h3[@class="code-header in-band"]' 'impl Escape<r#"<script>alert("Escape");</script>"#>'
 impl Escape<r#"<script>alert("Escape");</script>"#> {
     pub fn f() {}
 }
```

Reply:
My process was this: I went into the HTML and copied the ID, making sure to copy it from the `id=` field, not the `href=` field. Also, I got it from the raw HTML, not the Dev Tools console.

Also, when debugging these XPath tests it's useful to know that the text content at the end is a substring match. So if you're getting match failures, you can set the text content to `''` and see if you get a match, then gradually add back text and see if it still passes.